Update readme with game description and controls (#147)
aws-lumberyard-dev/o3de-multiplayersample: README.md

# MultiplayerSample Project

> A simple third-person multiplayer sample for O3DE. Its lazer tag in space!

Reviewer: "laser". Well technically L.A.S.E.R.
Light 
Amplification by 
Stimulated 
Emission of 
Radiation

> A simple third-person multiplayer sample for O3DE. Its lazer tag in space!

Reviewer 2: ```suggestion
A simple third-person multiplayer sample for O3DE. It's lazer tag in space!
```

Just double-checking, is "lazer tag" (or "laser tag") trademarked at all?

> **_NOTE:_** For Linux, see the Linux specific setup in [README_LINUX](./README_LINUX.md).

## Gameplay Overview
This game sample has players competing to hit the highest score to win. Played over a series of rounds, each round brings greater opportunity for rewards but at higher risks.  

Additionally, players have a fixed amount of energy shields to protect them. Shooting other players damages their shields and getting hit damages your shields. Once your shields are depleted, you will respawn but at the cost of some of your collected gems.

Do you risk it all to win? 

Game features:
* 3rd-person shooting
* A configurable number of rounds; defaults to 3
* Different gem spawning patterns per round to drive exploration
* Support for 1 to 15 players (goal is to raise performance to support 25 players per level)
* Rich sounds and VFX support
* Teleporters to aid exploration
* Many points of extensibility

> Note: Theres no penalty for respawn if you are knocked off the level or use the teleporters

Reviewer: There's

> Note: Theres no penalty for respawn if you are knocked off the level or use the teleporters

Reviewer 2: ```suggestion
Note: There's no respawn penalty if you are knocked off the level or use the teleporters.
```

### Player Controls

* Move using: `WASD` 
* Sprint: Hold `shift`
* Jump: `space` 
* Look around: use mouse
* Fire primary: `left mouse button`
* See scoreboard: Hold `tab`

> * Open game menu: `ESC`

Reviewer 2: ```suggestion
* Open game menu: `esc`
```
(Everything else is using lower case for special keys, seems like this should too)

## Gameplay Overview
This game sample has players competing to hit the highest score to win. Played over a series of rounds, each round brings greater opportunity for rewards but at higher risks.  Players race to collect gems scattered all over the level.

Additionally, players have energy shields to protect them. Shooting other player damages their shields and getting hit damages your shields. Once your shields are depleted, you will respawn but at the cost of some of your collected gems.

Do you risk it all to win? 

Game features:
* 3rd-person character setup.
* Shooting with laser pistols.
* A configurable number of rounds; defaults to 3.
* Configurable gem spawning patterns per round to drive exploration.
* Support for 1 to 15 players (goal is to raise performance to support 25 players per level).
* Rich sounds and VFX support.
* Teleporters to aid exploration.
* Many points of extensibility.

Note: There's no penalty for respawn if you are knocked off the level or use the teleporters.

### Player Controls

* Move using: `w,a,s,d` 
* Sprint: Hold `shift`
* Jump: `space` 
* Look around: use mouse

> * Fire primary: `left mouse button`

Author: ```suggestion
* Fire primary weapon: `left mouse button`
```

* See scoreboard: Hold `tab`
* Open game menu: `esc`
* Draw/holster active weapon: `e`

To be implemented:

> * Fire secondary weapon: `right mouse button`

Reviewer 3: This should be fire primary weapon

Reviewer 4: I believe left mouse button is the primary weapon. Right mouse button isn't bound to anything and it would be secondary. So this is correct.

Reviewer 2: I think @mcphedar just marked the wrong line, I think he's saying that the "Fire primary" line should say "Fire primary weapon"

Reviewer 2: Or, never mind, just read his other comment. I'm wrong, though I think that would still be a good change...

#### Running the Client

The client launcher can be run with:

```shell
MultiplayerSample.GameLauncher.exe --console-command-file=launch_client.cfg
```

This will connect a client to the local server and start a multiplayer session.
For convenience, you can run launch_client.cmd (Windows) or launch_client.sh (Unix) directly.

#### Debugging in Visual Studio
When debugging MultiplayerSample.GameLauncher and MultiplayerSample.ServerLauncher from Visual Studio it's helpful to automatically host and connect; thereby avoiding having to open the console (~) once the application opens and explicitly executing the `host` and `loadlevel` command on server, or the `connect` command on client. For convenience, Gem/Code/CMakeLists.txt defines `ADDITIONAL_VS_DEBUGGER_COMMAND_ARGUMENTS` which allows Visual Studio to automatically populate the debugger with command arguments. 

By default, launch_client.cfg is used when debugging the GameLauncher and launch_server.cfg is used when debugging the ServerLauncher.

> > When debugging set `net_UdpTimeoutConnections` to false, this will prevent connection closures when stopped on breakpoints.

Reviewer: You're dropping all the pro-tips!

## Levels in this Project
This project ships with several levels, the ones of note are:

1. NewStarBase - The main game level, default to using this level
2. GamePlayTest - Everything needed for gameplay, but in a tiny, fast loading level. Gems, HUD, everything is there.
3. StartMenu - An example menu to enable joining, hosting and connecting to servers.
4. MultiplayerScriptingSample - An example of Multiplayer and scripting.

> Other levels in project are there for testing, performance evaluation purposes and can be ignored.

Reviewer 2: ```suggestion
Other levels in the project are there for testing or performance evaluation purposes and can be ignored.
```
